libs/template/docs/api/class-directive.md

```diff
@@ -0,0 +1,65 @@
+## ClassDirective
+
+The `rxClass` directive serves a convenient way of using observables for style (class) changes.
+
+Under the hood, it leverages a `RenderStrategy` which in turn takes care of optimizing the change detection
+of your component by passing wanted strategy to `rxClassStrategy`. The `ClassDirective` will apply style (class) changes to the template and manage change detection after it got an initial value.
+So if the incoming `Observable` emits its value lazily (e.g. data coming from `Http`), your style/class will be
+applied lazily as well. This can very positively impact the render performance of your application.
+
+### Problems with `async` and `ngClass` or `class`
+
+In Angular, a way of binding an observable to the `ngClass` or `class` directive could look like that:
+
+```html
+<div [ngClass]="{ 'visible': isVisible$ | async }"></div>
+<div [class.visible]="isVisible$ | async"></div>
+```
+
+When using `ngClass` or `class` directive with an Observable value and the `async` pipe in the template of an Angular application, the `async` pipe relies on the Zone to be present - it doesn't really trigger change detection by itself. It marks the component and its children as dirty waiting for the Zone to trigger change detection. So, in case you want to create a zone-less application, the `async` pipe won't work as desired.
```

Comment:
```suggestion
When using the native `ngClass` or `class` directive, the template of your component needs to get re-evaluated in order to apply the changes. In other words `ChangeDetection` has to run for your component in order to set a class to an `HTMLElement`.

With an Observable value you typically use something like the `async` pipe in order to reflect its values in your components template. ..TBD. ... more to continue, here is a link where the async behavior is already described. we can cross-link here: https://github.com/rx-angular/rx-angular/blob/main/libs/template/docs/performance-issues.md#binding-reactive-sources
```

Reply:
why resolved? I think the main benefit of `rxClass` is that it doesn't need changeDetection to apply it's changes. To get rid of the `async` pipe, you could use `push` or `rxLet` but you still need cd for applying class changes.